packages/data-context/src/sources/RelevantRunsDataSource.ts

```diff
@@ -180,31 +181,41 @@ export class RelevantRunsDataSource {
   /**
    * Clear the cached current run to allow the data source to pick the next completed run as the current
    */
-  //TODO figure out how to mock in test so do not need to pass in values here
-  async moveToNext (shasForTest?: string[]) {
+  async moveToNext (shas: string[]) {
     debug('Moving to next relevant run')
-    this.#currentRun = undefined
-    this.#currentCommitSha = undefined
 
-    const runs = await this.getRelevantRuns(this.ctx.git?.currentHashes || shasForTest || [])
+    await this.checkRelevantRuns(shas, true)
+  }
+
+  async checkRelevantRuns (shas: string[], clearCache: boolean = false) {
+    debug(`check relevant runs with ${shas.length} shas and clear cache set to ${clearCache}`)
+    if (clearCache) {
+      this.#currentRun = undefined
+      this.#currentCommitSha = undefined
+    }
+
+    const runs = await this.getRelevantRuns(shas)
```

Comment:
Should we put a `try/catch` here in case something in `getRelevantRuns` throws? I know you got ahead of the specific errors we saw with cloud connection, but not sure if other things could bubble up in some cases and lead to the same crashing issue? Looking at that function it seems very unlikely to throw.

Reply:
I think this does not need a try/catch now based on how the error is resolved.